pkg/login/social/social.go

```diff
@@ -244,6 +244,7 @@ type SocialConnector interface {
 	UserInfo(client *http.Client, token *oauth2.Token) (*BasicUserInfo, error)
 	IsEmailAllowed(email string) bool
 	IsSignupAllowed() bool
+	GetCustomAuthParams() []oauth2.AuthCodeOption
```

Comment:
Instead of adding a new function for each provider they could Implement AuthCodeURL and just append the specific options there?